### PR TITLE
Overview page fixes

### PR DESCRIPTION
**Pull request by sennett**
# Overview page fixes
  
## Changes 👷‍♀️

- [Includes liquidity on overview page fro DPM positions.](https://app.shortcut.com/oazo-apps/story/7209/overview-liquidity-is-not-complete)
- [Shows AAVE positions correctly on overview page](https://app.shortcut.com/oazo-apps/story/7263/bug-routing-issue-with-aave-from-my-positions-page).  Consolidates aaveDpmPositions and aavePositions into one observable.
  
## How to test 🧪

- open positions overview page with an AAVE stETH/ETH postiion.
- ensure that liquidity is displayed and the correct format and value
- ensure that multiply positions created using DS Proxy show as multiply positions on overview page

### CODE REVIEW

File: features/vaultsOverview/pipes/positions.ts [hunk 1]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
+            user: walletAddress,
+            proxy: dsProxyAddress,
+            vaultId: walletAddress,
+          },
+        ]
+      }
+
+      return readPositionCreatedEvents$(walletAddress).pipe(
+        map((positionCreatedEvents) => {
+          return [...positionCreatedEvents, ...dsFakeEvent]
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:171]: what happens when a user creates a position right now with DSProxy? We have those events so will we have this position two times?

File: features/vaultsOverview/pipes/positions.ts [hunk 2]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
               })
             },
           )
 
-          const tickerPrices = positionCreatedEvents.map(({ collateralToken, debtToken }) => {
-            const collateral = getTokenFromAddress(context, collateralToken)
-            const debt = getTokenFromAddress(context, debtToken)
-            return tickerPrices$([collateral, debt])
-          })
+          const tickerPrices = positionCreatedEvents.map(
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:188]: a lot of observables...
```
const allTokens = positionCreatedEvents.map((positionCreatedEvent) => {
            return [
              positionCreatedEvent.collateralTokenSymbol,
              positionCreatedEvent.debtTokenSymbol,
            ]
          }).flatMap((tokens) => tokens)
          
          const distinctTokens = Array.from(new Set(allTokens))
          const tickerPrices = tickerPrices$(distinctTokens)
  ```

File: features/vaultsOverview/pipes/positions.ts [hunk 3]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
+            ({ collateralTokenSymbol, debtTokenSymbol }) => {
+              return tickerPrices$([collateralTokenSymbol, debtTokenSymbol])
+            },
+          )
 
           const wrappedGetAaveReserveDataObservableList = positionCreatedEvents.map(
-            ({ debtToken }) => {
-              const debt = getTokenFromAddress(context, debtToken)
-              return wrappedGetAaveReserveData$(debt)
+            ({ debtTokenSymbol }) => {
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:195]: what about duplicates? User can have two position with the same debt token

File: features/vaultsOverview/pipes/positions.ts [hunk 4]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
+              return wrappedGetAaveReserveData$(debtTokenSymbol)
             },
           )
 
+          const liquidityObservables = positionCreatedEvents.map(({ debtTokenSymbol }) => {
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:200]: Literally the same filtering. Please do that in one `map`

File: features/vaultsOverview/pipes/positions.ts [hunk 5]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
+              ([
+                protocolDataList,
+                assetPricesList,
+                tickerPrices,
+                preparedAaveReserveData,
+                liquidities,
+                positionCreatedEvents,
+              ]) => {
+                const positions = protocolDataList.map(
+                  (
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:224]: come on, it's almost 100 lines in one `map` function. Please make a named function from it.

File: features/vaultsOverview/pipes/positions.ts [hunk 6]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
-                    type: mappymap[positionCreatedEvents[index].positionType],
-                  }
-
-                  return position
-                },
-              )
-            }),
+                )
+
+                return positions
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:312]: It's redundant. Do that without creating `positions` variable.

File: features/vaultsOverview/pipes/positions.ts [hunk 7]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
+                assetPricesList,
+                tickerPrices,
+                preparedAaveReserveData,
+                liquidities,
+                positionCreatedEvents,
+              ]) => {
+                const positions = protocolDataList.map(
+                  (
+                    {
+                      position: {
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:226]: It's not readable. Make it `map((protocolData, index) => ....`

File: features/vaultsOverview/pipes/positions.ts [hunk 8]
```diff
@@ -179,130 +129,189 @@ export function createAaveDpmPosition$(
...
+
+                    if (!positionId) {
+                      throw new Error(
+                        `Could not find position ID for proxy address ${positionCreatedEvents[index].proxyAddress}`,
+                      )
+                    }
+
+                    const isDebtZero = debt.amount.isZero()
+
+                    const tickerCollateralTokenPriceInUsd = tickerPrices[index][collateral.symbol]
```
Comment (jakubswierczek) [inline features/vaultsOverview/pipes/positions.ts:247]: image

A lot of getting elements by index. It's not a good approach. We should merge those arrays into one to do simple transforming

**Review by jakubswierczek — CHANGES_REQUESTED** [review]

File: features/aave/services/readPositionCreatedEvents.ts [hunk 9]
```diff
@@ -18,7 +27,7 @@ export function createReadPositionCreatedEvents$(
   context$: Observable<Context>,
   userDpmProxies$: (walletAddress: string) => Observable<UserDpmProxy[]>,
   walletAddress: string,
-): Observable<Array<PositionCreatedEventPayload>> {
+): Observable<Array<PositionCreated>> {
   return combineLatest(context$, userDpmProxies$(walletAddress)).pipe(
     switchMap(([context, dpmProxies]) => {
       return combineLatest(
```
[the remark this answers is not in the document]
Comment (sennett) [reply features/aave/services/readPositionCreatedEvents.ts:33]: This also resolves it as it's doing the same thing - I have moved to this.  What's the purpose of this contractV2?

File: features/vaultsOverview/pipes/positions.ts (hunk 1 above)
Comment (sennett) [reply features/vaultsOverview/pipes/positions.ts:171]: Good spot.  Resolved.

File: features/aave/services/readPositionCreatedEvents.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (jakubswierczek) [reply features/aave/services/readPositionCreatedEvents.ts:33]: `contractsv2` uses `ethers` instead of `web3`

**Review by jakubswierczek — APPROVED** [review]

**Pull request merged**